pkg/controller/endpointslice/topologycache/topologycache.go

```diff
@@ -132,6 +132,7 @@ func (t *TopologyCache) AddHints(si *SliceInfo) ([]*discovery.EndpointSlice, []*
 // cache.
 func (t *TopologyCache) SetHints(serviceKey string, addrType discovery.AddressType, allocatedHintsByZone EndpointZoneInfo) {
 	if len(allocatedHintsByZone) == 0 {
+		klog.V(2).Infof("No hints allocated for zones, removing them from %s EndpointSlices for %s Service", addrType, serviceKey)
```

Comment:
this will keep logging, shouldn't be better to add the log inside RemoveHints, only if it is actually removed?

Reply:
nevermind, this log is consistent